de.fhdo.lemma.service.open_api/src/de/fhdo/lemma/service/open_api/LemmaTechnologySubGenerator.xtend

```diff
@@ -0,0 +1,181 @@
+ */
+class LemmaTechnologySubGenerator {
+    /** Factory to actually create and manipulate a LEMMA Technology */
+    val TECHNOLOGY_FACTORY = TechnologyFactory.eINSTANCE
+    /** Factory to actually create and manipulate a LEMMA DataModel */
+    val DATA_FACTORY = DataFactory.eINSTANCE
+    /** Predefined instance of the <strong>TechnologyModel</strong>. This instance is populated
+     * with the various technologies from the OpenAPI model, e.g., media types.
+     */
+    val myTechModel = TECHNOLOGY_FACTORY.createTechnology
+    /** OpenAPI schema which will be used as source for generation. */
+    OpenAPI openApi
+    /** SLF4j Logger */
+    val static logger = LoggerFactory.getLogger(LemmaServiceSubGenerator);
+    /** Location where the generated file is written */
+    String targetFolder
+    /** List of all unique media types. */
+    val mediaTypes = <String>newHashSet
+
+    new(OpenAPI api, String genPath, String techFilename) {
+        super()
+        logger.debug("Creating new Technology Sub Generator...")
+        this.openApi = api
+        this.targetFolder = genPath + techFilename
+    }
+
+    def Technology generate() {
+        logger.debug("Starting generation...")
+        logger.debug("Initializing model instance...")
+        initilize()
+        logger.debug("...Technology Model initialized!")
+        logger.debug("Creating protocols...")
+        createProtocols()
+        logger.debug("...Data Structures created!")
+
+        if (OpenApiUtil.writeModel(myTechModel, targetFolder)) {
+            logger.info("Technology model generation successful!")
+            logger.info('''Model written to «targetFolder»''')
+        } else
+            throw new Exception("Technology model generation failed :(")
+        return myTechModel
+    }
+
+    def createProtocols() {
+        //creates "rest" as the default openapi protocol
+        val rest = TECHNOLOGY_FACTORY.createProtocol
+        rest.name = "rest"
+        openApi.paths.forEach[path, pathSchema|searchMediaTimes(pathSchema)]
+        mediaTypes.forEach [ type |
+            val df = TECHNOLOGY_FACTORY.createDataFormat
+            df.formatName = type
+            rest.dataFormats.add(df)
+        ]
+        //Sets "application/json" as default format if the OpenAPI desc entails it at all.
+        // If not, the first encountered media type is set as default.
+        rest.defaultFormat = rest.dataFormats.findFirst[
+            it.formatName.equals("application/json")
+        ] ?: rest.dataFormats.get(0)
+        myTechModel.protocols.add(rest)
+    }
+
+    def searchMediaTimes(PathItem item) {
+        val httpVerbList = <Operation>newArrayList
+        httpVerbList.addAll(item.get, item.delete, item.put,
+            item.post, item.patch, item.head, item.options
+        )
+        httpVerbList.forEach [ verb |
+            if (verb !== null) {
+                // Getting MediaTypes from responses
+                verb.responses?.forEach[
```

Comment:
Suggestion: Factor out in a helper method to better communicate the intent:
```
findMediaTypesFromResponses(Responses responses) {
    if (responses === null)
        return emptySet()

    val responseSchemata = responses.filter[responseSchema !== null].map[responseSchema].flatten
    val contents = responseSchemata.filter[content !== null].map[content].flatten.toSet
    return contents
}

mediaTypes.addAll(findMediaTypesFromResponses(responses))
```
Sure, it's more verbose helper-wise, but at the point of calling it communicates what is going on without the caller being affected by the complexity of Elvis operators and "null-aware" methods like `addToMediaTypes`.